pallets/attestation/src/lib.rs

```diff
@@ -154,17 +168,14 @@ pub mod pallet {
 	#[pallet::getter(fn attestations)]
 	pub type Attestations<T> = StorageMap<_, Blake2_128Concat, ClaimHashOf<T>, AttestationDetails<T>>;
 
+	// TODO: Migrate... Either delete or rename
 	/// Delegated attestations stored on chain.
 	///
 	/// It maps from a delegation ID to a vector of claim hashes.
 	#[pallet::storage]
-	#[pallet::getter(fn delegated_attestations)]
-	pub type DelegatedAttestations<T> = StorageMap<
-		_,
-		Blake2_128Concat,
-		DelegationNodeIdOf<T>,
-		BoundedVec<ClaimHashOf<T>, <T as Config>::MaxDelegatedAttestations>,
-	>;
+	#[pallet::getter(fn external_attestations)]
+	pub type ExternalAttestations<T> =
+		StorageDoubleMap<_, Twox64Concat, AuthorizationIdOf<T>, Blake2_128Concat, ClaimHashOf<T>, bool, ValueQuery>;
```

Comment:
Regarding the question whether we want to have an additional map `ClaimHash --> AuthorizationId`:

Is there any other use-case for `ExternalAttestations ` other than distinguishing between internal and external attestations so far? 

For now I don't see a use-case to store the second map, especially as we did not need it before. However, it will be expensive to add later by iterating over the ExternalAttestations maps.

Reply:
> The main point of having a map for delegated attestations was/is to make it easier, when a delegation is revoked, to revoke, if needed, all attestations issued with that specific delegation.

This is a big point. But looking up, which attestations where issued by an delegation node or similar might even be a bigger use case in the future.
Having the storage in the attestation pallet doesn't mean that it's inaccessible for the delegating pallet.
Regarding the cascading deletion. I think we should just fail in that case. These cascading things are tough to weight and should rather be handled by the SDK/UI.

> I would store any information in the VCO and not in the attestation pallet, which otherwise becomes a big pot of everything.

I disagree with the big pot of everything. The `ExternalAttestations` is merely an index for the `Attestation` store. Therefore it should life in the attestation pallet. If we would start to split that across pallets, there would be X different indices that need to be kept up to date.

Reply:
> Regarding the cascading deletion. I think we should just fail in that case. These cascading things are tough to weight and should rather be handled by the SDK/UI.

The operation is not meant to be triggered automatically on deletion, but the original attester (or a delegator) can lookup the attestation hashes, batch them, and revoke/remove them. And I am just reporting what I was told back then, since I also questioned the utility of this additional map. Now it makes definitely sense to have it, but it was never meant to be used for "cleaning up" automatically when a delegation is removed.

> I disagree with the big pot of everything. The ExternalAttestations is merely an index for the Attestation store.

All the attestations are still stored in the attestation pallet, so in my opinion it makes sense that each "authorisable" pallet would keep its own index, especially if something additional is needed or something changes. This would be easier if the access control trait would expose hooks for main operations, which it does not right now and would require further changes. But of course this is just my point of view.

Reply:
I still don't see why each pallet should keep their own index. That just makes it more complicated. 🤷 Some pallets might then also choose to not have the index? The code to keep track of the attestations needs to be duplicated for all the other pallets too.

If we want to have the hooks or not is a separate question. And I think we will need them later when we have more advanced use cases.

Reply:
Yes, that for me is one more point: pallets might not need to have an index at all. And I agree that it makes the code more complicated, and that why I am saying that the current solution also works (and can always be migrated in the future, if needed).

Reply:
I would guess that all pallets need an index. But i think only time will tell. :D